src/helpers.c

```diff
@@ -696,7 +696,7 @@ int get_dirfd_path(const char *fullname)
 		 * flag an error); otherwise, it can be safely deleted. Until
 		 * that check is implemented, always skip the file, because we
 		 * cannot safely determine if it can be deleted. */
```

Comment:
Can you remove this FIXME comment, or rewrite it as a normal code comment that describes the special handling? As currently written, I think it is irrelevant now :-)

Reply:
but isn't this FIXME still valid? We are only skipping the file if we find it to be not-safe to be deleted. But the FIXME talks about checking to see if real_path is marked non-deleted in the consolidated manifest, and deleting it if it can be safely deleted. After all the fix in this commit is only for not flagging an error.

Reply:
No, after some recent discussions, I don't think my original thinking behind this FIXME was correct... But fair point that the intent of your PR is to not flag an error, so the FIXME can be addressed separately.

Reply:
I'm ok with changing the message, I just wanted to understand the reasoning why :)

Reply:
Okay, this is my current thinking.

If `verify` detects the "not safe to delete" condition, the canonical absolute path either resides (a) outside the /usr tree (or under /usr/local), or (b) within the /usr tree (excluding /usr/local). For case (a), the FIXME is not relevant, because the canonical absolute path will not be present in the consolidated manifest due to the path not being tracked by swupd. For case (b), the FIXME is also not relevant, because the canonical absolute path *will* be present in the consolidated manifest, and `verify` will therefore operate on it in the normal course of iterating through the consolidated manifest. There is no need to check the consolidated manifest "out of order", as the FIXME implies, because `verify` will already fix everything it needs to.